pages/api/applications/[applicationId]/tasks/[taskId].ts

```diff
@@ -1,5 +1,116 @@
+import { PrismaClient, Task } from '@prisma/client';
 import type { NextApiRequest, NextApiResponse } from 'next';
+import { ApiResponse, StatusMessageType } from '../../../../../types/apiResponse';
+import { TaskData, TaskPatchData } from '../../../../../types/task';
+import { withAuthUser } from '../../../../../utils/auth/jwtHelpers';
+import { isValidDate } from '../../../../../utils/date/validations';
+import { createJsonResponse, HttpMethod, HttpStatus, rejectHttpMethod } from '../../../../../utils/http/httpHelpers';
+import { withPrismaErrorHandling } from '../../../../../utils/prisma/prismaHelpers';
 
-export default function handler(req: NextApiRequest, res: NextApiResponse) {
-  res.status(200).end(`Request sent to ${req.url}\n`);
+const prisma = new PrismaClient();
+
+enum MessageType {
+  INVALID_TASK_ID,
+  TASK_NOT_FOUND,
+  INVALID_APPLICATION_ID,
+  TASK_DELETED,
+  TASK_PATCHED,
+  INVALID_TITLE,
+  INVALID_DUE_DATE,
+  INVALID_NOTIFICATION_DATETIME,
+  INVALID_IS_DONE,
+}
+
+const messages = Object.freeze({
+  [MessageType.INVALID_TASK_ID]: { type: StatusMessageType.ERROR, message: 'Task id is invalid.' },
+  [MessageType.INVALID_APPLICATION_ID]: { type: StatusMessageType.ERROR, message: 'Application id is invalid.' },
+  [MessageType.TASK_NOT_FOUND]: { type: StatusMessageType.ERROR, message: 'Task cannot be found.' },
+  [MessageType.TASK_DELETED]: { type: StatusMessageType.SUCCESS, message: 'The task has been successfully deleted.' },
+  [MessageType.TASK_PATCHED]: { type: StatusMessageType.SUCCESS, message: 'The task has been successfully patched.' },
+  [MessageType.INVALID_TITLE]: { type: StatusMessageType.ERROR, message: 'The task title is invalid.' },
+  [MessageType.INVALID_DUE_DATE]: { type: StatusMessageType.ERROR, message: 'The task due date is invalid.' },
+  [MessageType.INVALID_NOTIFICATION_DATETIME]: {
+    type: StatusMessageType.ERROR,
+    message: 'The task notification date is invalid.',
+  },
+  [MessageType.INVALID_IS_DONE]: { type: StatusMessageType.ERROR, message: 'The task completion status is invalid.' },
+});
+
+async function handler(uid: string, req: NextApiRequest, res: NextApiResponse<ApiResponse<TaskData>>) {
+  const { taskId, applicationId } = req.query as { taskId: string; applicationId: string };
```

Comment:
```suggestion
  const { taskId, applicationId } = req.query as TaskRouteParams;
```

Should we define the type of the params in `types/user.ts`?